includes/title-meta.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Page Title Meta setup
+ *
+ * @package Go\TitleMeta
+ */
+
+namespace Go\TitleMeta;
+
+/**
+ * Set up Title Meta hooks
+ *
+ * @return void
+ */
+function setup() {
+	$n = function( $function ) {
+		return __NAMESPACE__ . "\\$function";
+	};
+
+	add_action( 'add_meta_boxes', $n( 'page_title_add_meta_boxes' ) );
+	add_action( 'save_post', $n( 'page_title_meta_box_data' ) );
+	add_filter( 'go_page_title_args', $n( 'hide_page_title' ) );
+}
+
+/**
+ * Hide page title box
+ *
+ * @param title_data $title_data The title data.
+ * @return title_data $title_data Filtered title data.
+ */
+function hide_page_title( $title_data ) {
+	if ( ! is_singular( 'page' ) || ! get_post_meta( get_the_ID(), '_page_title', true ) ) {
+		return $title_data;
+	} else {
+		$title_data['title'] = '';
+		return $title_data;
+	}
+}
+
+/**
+ * Add meta box
+ *
+ * @param post $post The post object.
+ * @link https://codex.wordpress.org/Plugin_API/Action_Reference/add_meta_boxes
+ */
+function page_title_add_meta_boxes( $post ) {
+	add_meta_box(
+		'page_title_checkbox',
+		__( 'Hide page title', 'go' ),
+		'Go\TitleMeta\\page_title_build_metabox',
+		'page',
+		'side',
+		'high'
+	);
+
```

Comment:
You can remove the space here so all methods don't have an empty space at the beginning and end of them.